### PR TITLE
Define emitter operator as bridge from stanza pipeline to receiver

### PR DESCRIPTION
**Pull request by djaglowski**
**Description:**
This PR adds a stanza "operator", which I've named "emitter". The purpose of this operator is to provide a mechanism for transmitting logs from the stanza pipeline to the stanza receiver. 

To explain this another way, consider that stanza has its own pipeline, which is made up of operators. This pipeline is very much analogous to OT's pipelines. This new operator serves a similar purpose to an OT exporter, in that it is the final step in the stanza pipeline.

Users will not need to be aware of the emitter (will not need to explicitly include it in their stanza pipeline) because it is being set as the default output operator, via the `WithDefaultOutput(emitter)` line in `factory.go`.

This PR is a continuation of #1096 (also see #1092 for additional context). The reason I've chosen this subset of functionality for the next PR in this effort, is that the receiver implementation will depend on receiving logs from the emitter. Although this functionality is a bit of an aside, I think it's worth considering before the receiver implementation.

**Testing:**
Emitter operator is fully unit tested.

### CODE REVIEW

**Comment by djaglowski** [comment]
This PR has effectively been reviewed as part of #1125.

What is the next step to getting this merged?

**Comment by tigrannajaryan** [comment]
> This PR has effectively been reviewed as part of #1125.
> 
> What is the next step to getting this merged?

Have all the changes been applied and all comments resolved? @keitwb we will need an approval of this to get merged.

**Review by keitwb — APPROVED** [review]

**Comment by djaglowski** [comment]
> Have all the changes been applied and all comments resolved? @keitwb we will need an approval of this to get merged.

The review of #1125 did not result in any changes to the code in this PR, so it should be good to go unless there is additional feedback.

**Comment by djaglowski** [comment]
@tigrannajaryan I believe this PR is all set, pending your approval.

File: receiver/stanzareceiver/receiver.go
Comment (tigrannajaryan) [inline receiver/stanzareceiver/receiver.go:28]: This doesn't seem to be used anywhere.

**Review by tigrannajaryan — APPROVED** [review]
LGTM, let's see what comes next. I believe this doesn't produce any logs yet, right?

File: receiver/stanzareceiver/receiver.go
Comment (djaglowski) [reply receiver/stanzareceiver/receiver.go:28]: That's true. Wasn't strictly necessary in this PR. It'll be used in the next one, which will be ready shortly.

**Comment by djaglowski** [comment]
> I believe this doesn't produce any logs yet, right?

That's correct. #1124 will have stanza actually running, and #1125 will add conversion of logs from stanza format.

Thanks for the review Tigran.

**Pull request merged**
